File: src/phorge/api/endpoints/git.py
"""Git repository API endpoints."""

from __future__ import annotations

from typing import Any

from phorge.api.client import ForgeClient


class GitAPI:
    def __init__(self, client: ForgeClient) -> None:
        self._client = client

    async def install(
        self,
        server_id: int,
        site_id: int,
        *,
        provider: str,
        repository: str,
        branch: str = "main",
        composer: bool = True,
    ) -> None:
        payload: dict[str, Any] = {
            "provider": provider,
            "repository": repository,
            "branch": branch,
            "composer": composer,
        }
        await self._client.post(
            f"/servers/{server_id}/sites/{site_id}/git", json=payload
        )

    async def update(
        self, server_id: int, site_id: int, *, branch: str
    ) -> None:
        await self._client.put(
            f"/servers/{server_id}/sites/{site_id}/git",
            json={"branch": branch},
        )

    async def remove(self, server_id: int, site_id: int) -> None:
        await self._client.delete(f"/servers/{server_id}/sites/{site_id}/git")

File: internal/forge/git.go
package forge

import (
	"context"
	"fmt"
	"net/http"
)

// Install installs a Git repository on a site.
func (s *GitService) Install(ctx context.Context, serverID, siteID int64, provider, repo, branch string, composer bool) error {
	body := map[string]any{
		"provider":   provider,
		"repository": repo,
		"branch":     branch,
		"composer":   composer,
	}
	path := fmt.Sprintf("/servers/%d/sites/%d/git", serverID, siteID)
	return s.client.do(ctx, http.MethodPost, path, body, nil)
}

// UpdateBranch changes the deployed branch for a site.
func (s *GitService) UpdateBranch(ctx context.Context, serverID, siteID int64, branch string) error {
	body := map[string]string{"branch": branch}
	path := fmt.Sprintf("/servers/%d/sites/%d/git", serverID, siteID)
	return s.client.do(ctx, http.MethodPut, path, body, nil)
}

// Remove removes Git integration from a site.
func (s *GitService) Remove(ctx context.Context, serverID, siteID int64) error {
	path := fmt.Sprintf("/servers/%d/sites/%d/git", serverID, siteID)
	return s.client.do(ctx, http.MethodDelete, path, nil, nil)
}
